recipe/meta.yaml

```
@@ -37,12 +35,32 @@ requirements:
     - pyprind >=2.9.3
     - py_stringmatching >=0.2.1
 
+{% set tests_to_ignore = "\"" %}
+# test_valid_join is called three times and results in, inter alia:
+# E       fixture 'scenario' not found
+# >       available fixtures: cache, capfd, capfdbinary, caplog,
+#           capsys, capsysbinary, doctest_namespace, monkeypatch,
+#           pytestconfig, record_property, record_testsuite_property,
+#           record_xml_attribute, recwarn, tmp_path, tmp_path_factory,
+#           tmpdir, tmpdir_factory
+# >       use 'pytest --fixtures [testpath]' for help on them.
+{% set tests_to_ignore = tests_to_ignore + "not test_valid_join " %}
+{% set tests_to_ignore = tests_to_ignore + "\"" %}
```

```suggestion
{% set tests_to_ignore = tests_to_ignore + "\"" %}
```
Is this test only being skipped on windows? I'm curious why the `\` instead of `/`

---

Those are used for escaping the quotation marks rather than being part of a path.

---

I agree that is hard to read: `\"`.

It is the escape for the double quotes that wrap the `not test_valid_join`. There is another one at line 38.

---

Is this preferred over `pytest -v py_stringsimjoin/tests --ignore=tests/test_valid_join`? It just seems like a lot of extra lines for one skipped test.

---

The extra lines are for the jinja variables, it's a way to keep things clean.

`pytest` way to skip things:

- `--ignore` ignores the whole file
- `-k` select or deselect a test using a pattern (e.g. `-k "not this_slow_test"` will skip `this_slow_test` and run everything else, and `-k "this_particular_test"` will run _only_ `this_particular_test` and skip the rest).
- `--deselect` - skip a test specified by its whole path

when you have too many of these (`--ignore`, and `--deselect` can be used multiple times and each for 1 file/test), it's nice to keep them sorted. Also, this way you could use selectors while adding values to the variable `tests_to_ignore` (which should be `tests_to_skip` in this case).

See for example:

https://github.com/AnacondaRecipes/zict-feedstock/pull/4/files#diff-f3725a55bf339595bf865fec73bda8ac99f283b0810c205442021f29c06eea9aR30-R33

---

yes maybe for 1 test we would not need all of this ehehe but maybe @ifitchet was ready with a template to skip more than 1 test and left the scaffolding there. It's ok for future usage.

---

Ha!  In this instance I _did_ start with a simple `-k something` but I _thought_ that something else broke so expanded it up into the full J2-style for more than one then realised I only needed the one (but left the J2-style code in place).

In particular, to @Jrice1317 , originally the code was, effectively, `-k 'something'` but it turns out that Windows doesn't like that kind of quoting and wants `-k "something"` and so to get a real `"` into an already "-ed string means you need the ugly-but-works `"\""` form.  (You get used to it!)